src/sources/audiosourceoggvorbis.cpp

```diff
@@ -0,0 +1,160 @@
+    if (!vi) {
+        qWarning() << "Failed to read OggVorbis file:" << getUrl();
+        return ERR;
+    }
+    setChannelCount(vi->channels);
+    setFrameRate(vi->rate);
+    if (0 < vi->bitrate_nominal) {
+        setBitrate(vi->bitrate_nominal / 1000);
+    } else {
+        if ((0 < vi->bitrate_lower) && (vi->bitrate_lower == vi->bitrate_upper)) {
+            setBitrate(vi->bitrate_lower / 1000);
+        }
+    }
+
+    ogg_int64_t pcmTotal = ov_pcm_total(&m_vf, kLogicalBitstreamIndex);
+    if (0 <= pcmTotal) {
+        setFrameCount(pcmTotal);
+    } else {
+        qWarning() << "Failed to read total length of OggVorbis file:" << getUrl();
+        return ERR;
+    }
+
+    return OK;
+}
+
+void AudioSourceOggVorbis::preDestroy() {
+    const int clearResult = ov_clear(&m_vf);
+    if (0 != clearResult) {
+        qWarning() << "Failed to close OggVorbis file" << clearResult;
+    }
+}
+
+AudioSource::diff_type AudioSourceOggVorbis::seekSampleFrame(
+        diff_type frameIndex) {
+    DEBUG_ASSERT(isValidFrameIndex(m_curFrameIndex));
+    DEBUG_ASSERT(isValidFrameIndex(frameIndex));
+
+    const int seekResult = ov_pcm_seek(&m_vf, frameIndex);
+    if (0 == seekResult) {
+        m_curFrameIndex = frameIndex;
+    } else {
+        qWarning() << "Failed to seek OggVorbis file:" << seekResult;
+        const ogg_int64_t pcmOffset = ov_pcm_tell(&m_vf);
+        if (0 <= pcmOffset) {
+            m_curFrameIndex = pcmOffset;
+        } else {
+            // Reset to EOF
+            m_curFrameIndex = getFrameIndexMax();
+        }
+    }
+
+    DEBUG_ASSERT(isValidFrameIndex(m_curFrameIndex));
+    return m_curFrameIndex;
+}
+
+AudioSource::size_type AudioSourceOggVorbis::readSampleFrames(
+        size_type numberOfFrames, sample_type* sampleBuffer) {
+    return readSampleFrames(numberOfFrames, sampleBuffer,
+            frames2samples(numberOfFrames), false);
+}
+
+AudioSource::size_type AudioSourceOggVorbis::readSampleFramesStereo(
+        size_type numberOfFrames, sample_type* sampleBuffer,
+        size_type sampleBufferSize) {
+    return readSampleFrames(numberOfFrames, sampleBuffer, sampleBufferSize,
+            true);
+}
+
+AudioSource::size_type AudioSourceOggVorbis::readSampleFrames(
+        size_type numberOfFrames, sample_type* sampleBuffer,
+        size_type sampleBufferSize, bool readStereoSamples) {
+    DEBUG_ASSERT(isValidFrameIndex(m_curFrameIndex));
+    DEBUG_ASSERT(getSampleBufferSize(numberOfFrames, readStereoSamples) <= sampleBufferSize);
+
+    const size_type numberOfFramesTotal = math_min(numberOfFrames,
+            size_type(getFrameIndexMax() - m_curFrameIndex));
+
+    sample_type* pSampleBuffer = sampleBuffer;
+    size_type numberOfFramesRemaining = numberOfFramesTotal;
+    while (0 < numberOfFramesRemaining) {
+        float** pcmChannels;
+        int currentSection;
+        const long readResult = ov_read_float(&m_vf, &pcmChannels,
```

Comment:
Be aware: long is different on 64 bit Linux and Windows. I do not see an issue here though.  
So please place a note, that this is an exception from: 
https://bugs.launchpad.net/mixxx/+bug/1094143